src/include/sof/audio/component.h

```diff
@@ -427,6 +427,38 @@ struct comp_ops {
 	 * @param data unBind info
 	 */
 	int (*unbind)(struct comp_dev *dev, void *data);
+
+	/**
+	 * Gets config in component.
+	 * @param dev Component device
+	 * @param param_id param id for each component
+	 * @param first_block first block of large block.
+	 * @param last_block last block of large block.
+	 * @param data_offset block offset.
+	 * @param data block data.
+	 * @return 0 if succeeded, error code otherwise.
+	 */
+	int (*get_large_config)(struct comp_dev *dev, uint32_t param_id,
+				bool first_block,
+				bool last_block,
+				uint32_t *data_offset,
```

Comment:
data_ptr ? an offset would not be a pointer but an int ?

Reply:
Host provides a buffer for FW to fill up.  FW will fill up some of them then tell host how much you can use by storing the value in data_offset. Here  *data_offset is for invoked get_large_config to store the usable size of config.

Reply:
Ok, I think I'm following. I assume you mean ```data_offset``` and possibly some others are filled in by the callee and not the caller ? If so, this should be documented in the comments.

Reply:
sure